cobc/codegen.c

```diff
@@ -9885,8 +9885,26 @@ output_report_control (struct cb_report *p, int id, cb_tree ctl, cb_tree nx)
 	x = CB_VALUE (ctl);
 	s = cb_code_field(x);
 	if(nx) {
-		output_report_control(p, id, nx, CB_CHAIN(nx));
+		output_report_control (p, id, nx, CB_CHAIN(nx));
 	}
+
+	bfound = 0;
+	for(i= p->num_lines-1; i >= 0; i--) {
+		if(p->line_ids[i]->report_control) {
+			struct cb_field *c = cb_code_field (p->line_ids[i]->report_control);
+			if(c == s) {
+				bfound = 1;
+				break;
+			}
+		}
+	}
+	if (!bfound) {
```

Comment:
This check, the warning and likely also the setting to null (or dropping the single unused control, as @lefessan pointed out) must be moved to typeck.c.

Reply:
Ok. I moved this